### DISCUSSION TITLE
duckdb in termux on Android

### DISCUSSION THREAD

**Discussion by zcatav** [duckdb/duckdb#13336] [General]
Hi,
Hi, I am using Duckdb in Termux on my Android phone. When I want to install sqlite support to open the sqlite database, I get the following error. Thanks in advance for your help in resolving it.

```
duckdb
v1.0.0 1f98600
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.

D install sqlite;

HTTP Error: Failed to download extension "sqlite_scanner" at URL "http://extensions.duckdb.org/v1.0.0/linux_arm64_android/sqlite_scanner.duckdb_extension.gz"

Extension "sqlite_scanner" is an existing extension.
```

**Comment by AnabasisXu** [comment]
I have the same problem. My guess is that `linux_arm64_android` does not exist yet.

**Comment by szarnyasg** [comment]
> I have the same problem. My guess is that linux_arm64_android does not exist yet.

Correct, we do not yet distribute the extensions for the `linux_arm64_android` platform.

You may try to build the extension manually from https://github.com/duckdb/sqlite_scanner but this will take some effort.

**Comment by abubelinha** [comment]
> Hi, I am using Duckdb in Termux on my Android phone. 

Would you mind sharing the steps you used to install Duckdb in Termux?
I was not succesful to do that yet.

I want to use it in Python.
I tried all this, but the last command fails:

```Python
$ pkg install -y duckdb
$ pkg install -y libduckdb
$ pkg install -y libduckdb-static
$ pip install duckdb
```


Thanks in advance
@abubelinha

**Comment by zcatav** [comment]
I installed Termux from F-droid market on Android. Then I updated the
system with 'pkg up' command. And I completed the installation with 'pkg
install duckdb' command. That's all.
Zeki Çatav
zekicatav.com

15 Nis 2025 Sal 02:05 tarihinde abubelinha ***@***.***> şunu
yazdı:

**Reply by abubelinha** [reply]
Thanks a lot @zcatav , that worked for me too.

Although that seems to install only the duckdb CLI, but not its Python package. 
I cannot import the python client from a termux python script (`import duckdb` fails because previous `pip install duckdb` freezes).
Do you know how to install it too @szarnyasg @Mause ?

Related: #13457 #5336

**Comment by zcatav** [comment]
Yes, this installation only installs duckdb as cli. You can try 'pip
install duckdb' command for Python.
Zeki Çatav
zekicatav.com

15 Nis 2025 Sal 11:06 tarihinde abubelinha ***@***.***> şunu
yazdı:
